ddtrace/tracer/span.go

```diff
@@ -256,6 +257,12 @@ func (s *span) setMetric(key string, v float64) {
 	}
 }
 
+// unsetMetric unsets a numeric tag, in our case called a metric. This method
+// is not safe for concurrent use.
+func (s *span) unsetMetric(key string) {
```

Comment:
We really don't need this. Calling `delete` in place is good enough and clear in behaviour.